plib/library/IpAddress.php

```diff
@@ -34,8 +34,17 @@ public static function getAvailable()
         throw new pm_Exception("Unable to get server IP: empty result.");
     }
 
+    public static function getPublic($ipAddress)
+    {
+        if (!isset(self::$_ipAddresses)) {
+            self::getAvailable();
+        }
+
+        return self::$_ipAddresses[$ipAddress];
+    }
+
     public static function getDefault()
     {
-        return reset(self::getAvailable());
+        return self::$_defaultIpAddress;
```

Comment:
Need to warmup cache with self::getAvailable().
And we have to return default IP even if nothing was marked as default in API.